src/main/java/world/bentobox/bentobox/database/objects/Island.java

```diff
@@ -940,4 +940,33 @@ public String getGameMode() {
     public void setGameMode(String gameMode) {
         this.gameMode = gameMode;
     }
-}
\ No newline at end of file
+    
+    /**
+     * Returns if this island has its end island generated.
+     * @return {@code true} if this island has its end island generated, {@code false} otherwise.
+     * @since 1.5.0
+     */    
+    public boolean hasEndIsland(){
+        if (!BentoBox.getInstance().getIWM().isEndGenerate(getWorld()) || !BentoBox.getInstance().getIWM().isEndIslands(getWorld()) || BentoBox.getInstance().getIWM().getEndWorld(getWorld()) == null ){
+            return false;   
+        }
+        Location l = this.getCenter();
+        l.setWorld(BentoBox.getInstance().getIWM().getEndWorld(l.getWorld()));
+        return l.getBlock().getType() == Material.BEDROCK;
```

Comment:
This check will work so long as there is a bedrock block at the island center location. That *should* be okay.